### PR TITLE
Merge `CODEOWNERS` Part 2

### PR DESCRIPTION
**Pull request by chevdor**
# Description

This PR:
- merge the `CODEOWNERS` from each former repos 
- add catch-alls and default owners

### CODE REVIEW

File: .github/CODEOWNERS
Comment (ggwpez) [inline .github/CODEOWNERS:25]: Is this the catch all for `cumulus`? Normally the locks review is only required for runtime files.  
Mabye use `core-devs` or so.

File: .github/CODEOWNERS
Comment (ggwpez) [inline .github/CODEOWNERS:29]: We will have a `bridges` folder soon, maybe you can already add it here https://github.com/paritytech/polkadot-sdk/pull/1392

File: .github/CODEOWNERS
Comment (chevdor) [reply .github/CODEOWNERS:25]: Yes, I wasn't 100% of those but I am sure we want at least a value. 
There is no **team** (in GH) for cumulus other than this one. I am totally fine with `core-devs` instead and I pinged @gilescope to see if we could provide more specific patterns below this catch-all to route review more accurately.

File: .github/CODEOWNERS
Comment (chevdor) [reply .github/CODEOWNERS:29]: Sure, happy to do so.

Comment (chevdor) [reply .github/CODEOWNERS:37]: I added that in the current PR as @serban300 is ooo atm and I don't want to rebase his PR "in his back" :)

File: .github/CODEOWNERS
Comment (chevdor) [reply .github/CODEOWNERS]: I **added** `core-devs` for now.
Keep in mind those are catch-all so we can add more specific patterns below that are more targetted in terms of content and people.

File: .github/CODEOWNERS
Comment (joepetrowski) [inline .github/CODEOWNERS:16]:
```suggestion
#   that handles might work better because they are more recognizable on GitHub
#   and you can use them for mentioning unlike an email.
```

**Review by coderobe — APPROVED** [review]

**Review by ggwpez — APPROVED** [review]

**Comment by ggwpez** [comment]
This actually closes https://github.com/paritytech/polkadot-sdk/issues/1164.  
There are more similar ones in the [`Polkadot-SDK (Repo)`](https://github.com/orgs/paritytech/projects/114) or [`Monorepo`](https://github.com/orgs/paritytech/projects/90) projects.

File: .github/CODEOWNERS
Comment (ordian) [inline .github/CODEOWNERS:28]: hmm, I think this was previously only needed for runtime files, which are going to be migrated to https://github.com/polkadot-fellows/runtimes

https://github.com/paritytech/polkadot-sdk/pull/1239#issuecomment-1697070397

Comment (chevdor) [reply .github/CODEOWNERS:28]: Are you suggesting using `core-devs` for `/polkadot` as catch-all as well ?

Comment (ordian) [reply .github/CODEOWNERS:28]: yeah, probably start with not to strict rules and enforce other rules via PCR

Comment (chevdor) [reply .github/CODEOWNERS:28]: Yes the idea here is not to be too strict but have "full" coverage for the relevant files (ie all files but things like `.gitignore`, etc...)

**Review by juangirini — APPROVED** [review]

File: .github/CODEOWNERS
Comment (bkchr) [inline .github/CODEOWNERS:30]:
```suggestion
```

**Review by bkchr — CHANGES_REQUESTED** [review]
Please stop introducing random rules that weren't present before.

**Review by sam0x17 — APPROVED** [review]

File: .github/CODEOWNERS
Comment (franciscoaguirre) [reply .github/CODEOWNERS:28]: This is definitely too strict

**Review by franciscoaguirre — COMMENTED** [review]
We could add a new rule `/polkadot/xcm/` and tag the XCM team, or Keith, Just and me

**Comment by chevdor** [comment]
@bkchr, the idea is to have some fallback/catch-all for:
- `/substrate`
- `/polkadot`
- `/cumulus`

Ideally, those catch-all never trigger because we have more accurate rules to cover the content of the folders mentioned above. They should be covered nonetheless in case `CODEOWNERS` rules end up not covering everything* (that can easily happen as the project grows and new content is added).

* with the current state of the PR, I did not try 100% coverage, but those top level folder should be covered.
This is why I would rather not keep those empty.

> We could add a new rule /polkadot/xcm/ and tag the XCM team, or Keith, Just and me

@franciscoaguirre Yes, totally and we should.
I think it is better to use teams but we may not have all those we needs defined. 
In your case with `xcm`, there is https://github.com/orgs/paritytech/teams/xcm. So if you 3 are fine, we can add that rule. Just let me know and I will add it or feel free to commit into the PR.

File: .github/CODEOWNERS
Comment (altonen) [inline .github/CODEOWNERS:55]:
```suggestion

# Networking-related code
/substrate/client/network @paritytech/networking 

```

**Comment by bkchr** [comment]
Why do we need these fallback catch all? For what exactly?
Who told you that we need this?

**Review by sam0x17 — APPROVED** [review]

**Comment by chevdor** [comment]
Having full coverage with the CODEOWNERS allows matching any relevant file to an [owner](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#about-code-owners) (or a set of ~).

`CODEOWNERS` is mainly used to define who should review but also helps notify people about relevant content. As a side effect, it helps figuring out who is the right contact for a given piece of code, documentation, script, etc... 
For regular contributors, it is often trivial and they know who to contact. Looking at the history of changes can also help but reflects only the past. For new contributors and automation however, a proper coverage via `CODEOWNERS` is very helpful.

Without a catch-all in place, we will only match those patterns that are defined for as long as they are valid. 

Currently we have quite such rules for substrate but not much for the rest and none of the code for polkadot 

If my initial proposal for the top level owner can be improved, please propose better choices but keep in mind that those are only cath-all and **should have no effect if we have good rules for the content**.

On that last point, I see already quite some suggestions in this thread but I recon it may take a while to make it complete.

I can split this PR into 2, in order to deal with #1164 first and deal with further discussions in a follow-up PR:
- simple PR with [what we currently have](https://github.com/paritytech/polkadot-sdk/pull/1415/commits/416445d081cbd23e30e1d6311da2149d93cc72b2) without the new patterns
- this PR that adds more rules

**Comment by franciscoaguirre** [comment]
@chevdor I added the XCM team to the file but it's showing an error.
It might be the XCM team needs write access to polkadot-sdk? Right now under `repositories` I only see `paritytech/xcm`

File: .github/CODEOWNERS
Comment (acatangiu) [inline .github/CODEOWNERS:67]:
```suggestion
/substrate/client/consensus/babe/ @andresilva
/substrate/client/consensus/beefy/ @acatangiu
```

**Comment by chevdor** [comment]
@franciscoaguirre 
> It might be the XCM team needs write access to polkadot-sdk? Right now under repositories I only see paritytech/xcm

yes indeed, this is mentioned [here](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#about-code-owners):

> The people you choose as code owners must have write permissions for the repository. When the code owner is a team, that team must be visible and it must have write permissions, even if all the individual members of the team already have write permissions directly, through organization membership, or through another team membership.

**Comment by chevdor** [comment]
I extracted the simple merge as https://github.com/paritytech/polkadot-sdk/pull/1472

**Comment by chevdor** [comment]
@franciscoaguirre no I don't, sorry

**Comment by bkchr** [comment]
> For regular contributors, it is often trivial and they know who to contact. Looking at the history of changes can also help but reflects only the past. For new contributors and automation however, a proper coverage via `CODEOWNERS` is very helpful.

Not sure how a catch all should be helpful to know who to contact. I'm against this pr. I don't see any reason for this. Having individual parts beinged covered by `CODEOWNERS` is fine, while this also already isn't working. People are getting pinged any mainly ignore it. I'm going to close this issue. If someone finds this ultra important, the discussion with a proper proposal and issue description can continue in an issue.

**Pull request closed without merging**
